src/main/java/org/zaproxy/zap/extension/jwt/JWTExtension.java

```diff
@@ -31,23 +49,87 @@
 public class JWTExtension extends ExtensionAdaptor {
 
     protected static final Logger LOGGER = Logger.getLogger(JWTExtension.class);
+    private static final List<Class<? extends Extension>> DEPENDENCIES;
+
+    static {
+        List<Class<? extends Extension>> dependencies = new ArrayList<>(1);
+        dependencies.add(ExtensionHttpFuzzer.class);
+        DEPENDENCIES = Collections.unmodifiableList(dependencies);
+    }
+
+    private HttpFuzzerHandler httpFuzzerHandler;
+    private JWTMessageLocationReplacerFactory jwtMessageLocationReplacerFactory;
+
+    @Override
+    public List<Class<? extends Extension>> getDependencies() {
+        return DEPENDENCIES;
+    }
 
     @Override
     public String getAuthor() {
         return "KSASAN preetkaran20@gmail.com";
     }
 
+    @Override
+    public void init() {
+        JWTI18n.init();
+        jwtMessageLocationReplacerFactory = new JWTMessageLocationReplacerFactory();
+        MessageLocationReplacers.getInstance()
+                .addReplacer(HttpMessage.class, jwtMessageLocationReplacerFactory);
+    }
+
+    @Override
+    public void initView(ViewDelegate view) {
+        super.initView(view);
+    }
+
     @Override
     public void hook(ExtensionHook extensionHook) {
         super.hook(extensionHook);
-        JWTI18n.init();
-        try {
-            extensionHook.addOptionsParamSet(getJWTConfiguration());
-            extensionHook.getHookView().addOptionPanel(new JWTOptionsPanel());
-            LOGGER.debug("JWT Extension loaded successfully");
-        } catch (Exception e) {
-            LOGGER.error("JWT Extension can't be loaded. Configuration not found or invalid", e);
-        }
+        MessagePanelManager panelManager =
+                Control.getSingleton()
+                        .getExtensionLoader()
+                        .getExtension(ExtensionFuzz.class)
+                        .getClientMessagePanelManager();
+        panelManager.addViewFactory(RequestAllComponent.NAME, new JWTFuzzPanelViewFactory(null));
```

Comment:
These (and the options panel) should be added only if there's view (there's a `hasView()` method), to avoid creating unnecessary view components in daemon and command line modes. Same check would apply in `unload`.

Reply:
done.